src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp

```diff
@@ -223,7 +225,11 @@ void Ekf::controlMagFusion(const imuSample &imu_sample)
 					// declination angle over time.
 					const bool update_all_states = _control_status.flags.mag_3D || _control_status.flags.mag_hdg;
 					const bool update_tilt = _control_status.flags.mag_3D;
-					fuseMag(mag_sample.mag, R_MAG, H, aid_src, update_all_states, update_tilt);
+					const bool fused = fuseMag(mag_sample.mag, R_MAG, H, aid_src, update_all_states, update_tilt);
```

Comment:
Should `update_all_states` require yaw_align? 

Or alternatively should yaw_align be a strict requirement for mag_3D and mag_hdg?